src/handler/http/models/dashboards.rs

```diff
@@ -0,0 +1,228 @@
+use config::meta::dashboards::{v1, v2, v3, v4, v5};
+use serde::{Deserialize, Serialize};
+use serde_json::Value as JsonValue;
+use utoipa::ToSchema;
+
+/// HTTP request body for the `CreateDashboard` endpoint.
+#[derive(Debug, Deserialize, ToSchema)]
+pub struct CreateDashboardRequestBody(JsonValue);
+
+/// HTTP response body for `CreateDashboard` endpoint.
+#[derive(Debug, Serialize, ToSchema)]
+pub struct CreateDashboardResponseBody(DashboardDetails);
+
+/// HTTP response body for `GetDashboard` endpoint.
+#[derive(Debug, Serialize, ToSchema)]
+pub struct GetDashboardResponseBody(DashboardDetails);
+
+/// HTTP request body for `UpdateDashboard` endpoint.
+#[derive(Debug, Deserialize, ToSchema)]
+pub struct UpdateDashboardRequestBody(JsonValue);
+
+/// HTTP response body for `UpdateDashboard` endpoint.
+#[derive(Debug, Serialize, ToSchema)]
+pub struct UpdateDashboardResponseBody(DashboardDetails);
+
+/// HTTP URL query component that contains parameters for listing dashboards.
+#[derive(Debug, Deserialize, utoipa::IntoParams)]
+#[into_params(style = Form, parameter_in = Query)]
+pub struct ListDashboardsQuery {
+    /// Optional folder ID filter parameter
+    ///
+    /// If neither `folder` nor any other filter parameter are set then this
+    /// will search for all dashboards in the "default" folder.
+    ///
+    /// If `folder` is not set and another filter parameter, such as `title`, is
+    /// set then this will search for dashboards in all folders.
+    folder: Option<String>,
+
+    /// The optional case-insensitive title substring with which to filter
+    /// dashboards.
+    title: Option<String>,
+}
+
+/// HTTP response body for `ListDashboards` endpoint.
+#[derive(Debug, Serialize, ToSchema)]
+pub struct ListDashboardsResponseBody {
+    pub dashboards: Vec<ListDashboardsResponseBodyItem>,
+}
+
+/// An item in the list returned by the `ListDashboards` endpoint.
+#[derive(Debug, Serialize, ToSchema)]
+pub struct ListDashboardsResponseBodyItem {
+    pub hash: String,
+    pub version: i32,
+
```

Comment:
Additional fields for folder ID, folder name, dashboard title, etc will be added here.